docs/content/users/basics/architecture.md

```diff
@@ -0,0 +1,156 @@
+
+`mutagen` directory
+: Contains `mutagen/mutagen.yml`, where you can [override the default Mutagen configuration](../install/performance.md#advanced-mutagen-configuration-options).
+
+`mysql` directory
+: Contains [optional `mysql` or `mariadb` configuration](../extend/customization-extendibility.md#providing-custom-mysqlmariadb-configuration-mycnf).
+
+`nginx` directory
+: Can be used for add-on [nginx snippets](../extend/customization-extendibility.md#custom-nginx-configuration).
+
+`nginx_full` directory
+: Contains the nginx configuration used by the web container, which [can be customized](../extend/customization-extendibility.md#providing-custom-nginx-configuration).
+
+`postgres` directory
+: Contains `postgres/postgresql.conf`, which can be edited if needed. Remove the `#ddev-generated` line at the top to take it over.
+
+`providers` directory
+: Contains examples and implementations to demonstrate how the [`ddev pull`](../basics/commands.md#pull) command can work with [hosting providers](../providers/index.md).
+
+`web-build` directory
+: Can be used to provide a [custom Dockerfile](../extend/customizing-images.md) for the web container.
+
+`xhprof` directory
+: Contains the `xhprof_prepend.php` file that can be used to customize [xhprof](../debugging-profiling/xhprof-profiling.md) behavior for different types of websites.
+
+#### Look But Don’t Touch
+
+Files beginning with `.` are hidden because they shouldn’t be fiddled with; most are regenerated, and thus overwritten, on every `ddev start`:
+
+`.dbimageBuild` directory
+: The generated Dockerfile used to customize the `db` container on first start.
+
+`.ddev-docker-compose-base.yaml`
+: The base docker-compose file used to describe a project.
+
+`.ddev-docker-compose-full.yaml`
+: The result of preprocessing `.ddev-docker-compose-base.yaml` using `docker-compose config`. Mostly it replaces environment variables with their values.
+
+`.gitignore`
+: The `.gitignore` is generated by DDEV and should generally not be edited or checked in. (It gitignores itself to make sure you don’t check it in.) It’s generated on every `ddev start` and will change as DDEV versions change, so if you check it in by accident it will always be showing changes that you don’t need to see in `git status`.
+
+`.global_commands`
+: Temporary directory used to get global commands available inside a project. You shouldn’t ever have to look there.
+
+`.homeadditions`
+: Temporary directory used to consolidate global `homeadditions` with project-level `homeadditions`. You shouldn’t ever have to look here.
+
+`.webimageBuild` directory
+: The generated Dockerfile used to customize the web container on first start.
+
+### Global Files
+
+There’s only one global `.ddev` directory, which typically lives in your home folder: `~/.ddev` or `$HOME/.ddev`.
```

Comment:
```suggestion
There’s only one global `.ddev` directory, which lives in your home directory: `~/.ddev` (`$HOME/.ddev`).
```

Reply:
Is there a standard way of referring to the home directory path? I’m used to seeing `~/.ddev`, which may be overly Mac/Linux-specific, and the parenthetical seems clunky here. If there’s a winning convention I’d love to be more consistent about it in the docs.

Reply:
Unfortunately, people just don't understand no matter what you say. And then they make big mistakes, like thinking that /home is their home directory. I'm open to better things as well, but have been burned many times by people not understanding the basic concept of "home directory".

Reply:
> Unfortunately, people just don't understand no matter what you say.

I’d like to take a stab at explaining it carefully and sticking to a convention, then!

It’s probably worth introducing the concept along with any other fundamentals and sticking to a convention for it. Something like:

> DDEV stores its global files in your home directory, which you may know as your user directory depending on what operating system you use. Throughout the documentation we’ll use the Linux+Mac convention of using `~` to represent that directory. You can use a terminal to navigate to that directory using `cd ~` on Linux or macOS and `cd %HOMEPATH%` on Windows. (...)

Is there a style of referring to the home directory that you’d prefer so the docs can be internally consistent? (Can’t possibly help a confused person to refer to it differently on various pages.)

Reply:
* FYI, just `cd` goes to homedir
* Explaining the convention `~` is a great thing.. but it won't help the 99% who don't read through.
* Most people (hopefully) on traditional Windows use git-bash, so hopefully `cd` or `echo $HOME` is good enough there as well.
* There are some people who use PowerShell. So %HOMEPATH% doesn't work for them. Oh the permutations! Few people use cmd.